src/plugins/legacy/polygonRoi/polygonEventFilter.cpp

```diff
@@ -68,145 +69,137 @@ void polygonEventFilter::updateView(medAbstractImageView *view)
-                    minDist = dist;
-                    closestManager = manager;
-                }
+                interactorStyleRepulsor->GetManager()->SetMasterRoi(true);
+                interactorStyleRepulsor->GetManager()->interpolateIfNeeded();
+                addRoisInAlternativeViews();
+                manageTick();
             }
         }
+    }
 
-        QList<QColor> colors = updateColors(colorsToExclude);
-        QMenu *labelMenu = updateLabelMenu(colors);
-        QMenu *roiManagementMenu = new QMenu("Delete");
-        if (closestManager && closestManager->getMinimumDistanceFromNodesToEventPosition(mousePos) < 10 )
-        {
-            mainMenu.addMenu(roiManagementMenu);
-            QAction *deleteOneNodeAction = new QAction("One node", roiManagementMenu);
-            connect(deleteOneNodeAction, &QAction::triggered, [=](){
-                emit nodeToDelete(closestManager, mousePos[0], mousePos[1]);
-            });
-            roiManagementMenu->addAction(deleteOneNodeAction);
-
-            QAction *deleteContour = new QAction("Contour", roiManagementMenu);
-            connect(deleteContour, &QAction::triggered, [=](){
-                emit contourToDelete(closestManager);
-            });
-            roiManagementMenu->addAction(deleteContour);
-
-            QAction *deleteLabel = new QAction("Label", roiManagementMenu);
-            connect(deleteLabel, &QAction::triggered, [=](){
-                emit labelToDelete(closestManager);
-            });
-            roiManagementMenu->addAction(deleteLabel);
-
-            QAction *saveMask = new QAction("Save Mask", &mainMenu);
-            connect(saveMask, &QAction::triggered, [=](){
-                emit maskToSave(closestManager);
-            });
-
-            mainMenu.addMenu(roiManagementMenu);
-            mainMenu.addAction(saveMask);
-        }
-        else
-        {
-            labelMenu->setTitle("Label");
-            mainMenu.addMenu(labelMenu);
-        }
+    return false;
+}
 
+bool polygonEventFilter::mousePressEvent(medAbstractView * view, QMouseEvent *mouseEvent)
+{
+    if ( ! activateEventFilter )
+        return false;
 
-        mainMenu.exec(mouseEvent->globalPos());
-        delete labelMenu;
-        delete roiManagementMenu;
+    medAbstractImageView *v = dynamic_cast<medAbstractImageView *> (view);
+    if ( !currentView )
+    {
+        return false;
+    }
+    if (currentView != v)
+        return false;
 
-        return true;
+    if (mouseEvent->button()==Qt::LeftButton)
+    {
+        return leftButtonBehaviour(view, mouseEvent);
+    }
+    else if (mouseEvent->button()==Qt::RightButton)
+    {
+        return rightButtonBehaviour(view, mouseEvent);
     }
     return false;
 }
 
+bool polygonEventFilter::updateMainViewOnChosenSlice(medAbstractImageView *view, QMouseEvent *mouseEvent)
+{
+    QPointF position;
+    position.setX(mouseEvent->x()*QGuiApplication::screenAt(mouseEvent->pos())->devicePixelRatio());
```

Comment:
maybe `QGuiApplication::screenAt(mouseEvent->pos())->devicePixelRatio()` could be computed once

Reply:
I think it's dangerous if you change screen on the fly

Reply:
I agree with florent. The goal is to detect if users unplugged their 2nd screen, or move the medInria window from one screen to another.

Reply:
oki, no problem